packages/kotti-ui/source/kotti-accordion/KtAccordion.vue

```diff
@@ -1,145 +1,138 @@
 <template>
-	<div class="accordion">
-		<div
-			class="accordion__header"
-			:class="{ 'accordion__header--clickable': isFullyClickable }"
-			@click="headerToggle"
-		>
-			<div class="accordion__title">
+	<div class="kt-accordion">
+		<div :class="headerClasses" @click="headerToggle">
+			<div class="kt-accordion__title">
 				<slot name="title">
-					<i v-if="icon" class="yoco accordion__title__icon" v-text="icon" />
-					<div class="accordion__title__text" v-text="title" />
+					<i v-if="icon" class="yoco kt-accordion__title__icon" v-text="icon" />
+					<div class="kt-accordion__title__text" v-text="title" />
 				</slot>
 			</div>
-			<div class="accordion__toggle" @click.stop="toggle">
-				<i class="yoco" v-text="isOpen ? 'minus' : 'plus'" />
+			<div class="kt-accordion__toggle" @click.stop="toggle">
+				<i class="yoco" v-text="toggleIcon" />
 			</div>
 		</div>
-		<div
-			class="accordion__content"
-			:class="isOpen ? 'is-open' : 'is-close'"
-			:style="`--height: ${contentHeight}px`"
-			@click="setHeight"
-		>
-			<div ref="contentInner" class="inner">
+		<div :class="contentClasses">
+			<div ref="contentInner" class="kt-accordion__content__inner">
 				<slot />
 			</div>
 		</div>
 	</div>
 </template>
 
-<script>
-export default {
+<script lang="ts">
+import { Yoco } from '@3yourmind/yoco'
+import { computed, defineComponent, ref } from '@vue/composition-api'
+
+import { useSlideAnimaion } from './hooks'
+import { KottiAccordion } from './types'
+
+export default defineComponent<KottiAccordion.PropsInternal>({
 	name: 'KtAccordion',
 	props: {
 		icon: { default: null, type: String },
 		isClosed: { default: false, type: Boolean },
 		isFullyClickable: { default: false, type: Boolean },
 		title: { required: true, type: String },
 	},
-	data() {
+	setup(props) {
+		const contentInner = ref<HTMLElement | null>(null)
+
+		const { isContentOpen, toggle } = useSlideAnimaion(contentInner, {
+			duration: 300,
+			isInitiallyClosed: props.isClosed,
+		})
 		return {
-			isOpen: true,
-			contentHeight: null,
-			interval: null,
+			contentClasses: computed(() => ({
+				'kt-accordion__content': true,
+				'kt-accordion__content--is-closed': !isContentOpen.value,
+				'kt-accordion__content--is-open': isContentOpen.value,
+			})),
+			contentInner,
+			headerClasses: computed(() => ({
+				'kt-accordion__header': true,
+				'kt-accordion__header--clickable': props.isFullyClickable,
+			})),
+			headerToggle: () => {
+				if (props.isFullyClickable) toggle()
+			},
+			toggle,
+			toggleIcon: computed(() =>
+				isContentOpen.value ? Yoco.Icon.MINUS : Yoco.Icon.PLUS,
+			),
 		}
 	},
-	mounted() {
-		this.isOpen = !this.isClosed
-		// This is the best solution right now
-		// It will handle the case when the content change
-		const reloadingTime = 500
-		this.interval = setInterval(this.setHeight, reloadingTime)
```

Comment:
Why is this no longer necessary?

Reply:
This one is debatable. What this does in the current version of the Accordion, now every time the content height of the content changes this will trigger an animation. I personally dislike both the visual feel of that an the implementation. Now a content height change is will be drawn immediately without any transition.

Change my mind

Reply:
IIRC, this code was never about any fancy animation or whatever. It was because the max height was previously specified due to the need to interpolate between 0 and height for the opening.

So, basically, without this code, when the content expands it just wouldn’t expand the accordion.